README.md

```diff
@@ -52,13 +54,15 @@ describe('router', () => {
     });
   });
 
-  it('next/link can be tested too', () => {
+  it('next/link can be tested too', async () => {
     render(<NextLink href="/example?foo=bar">Example Link</NextLink>);
     fireEvent.click(screen.getByText('Example Link'));
-    expect(router).toMatchObject({
-      asPath: '/example?foo=bar',
-      pathname: '/example',
-      query: { foo: 'bar' },
+    await waitFor(() => {
+      expect(router).toMatchObject({
+        asPath: '/example?foo=bar',
+        pathname: '/example',
+        query: { foo: 'bar' },
+      });
```

Comment:
Ok, before I move forward with my PR, I actually have an open question ... SHOULD this library try to simulate the async behavior of Next, or should it actually be SYNC, for ease of development?

For example, the synchronous test for clicking  this `next/link` was way simpler than the async test (requiring wrapping in `await waitFor`) ... and I imagine that there are MANY use-cases where sync would be better.

I guess my question is ... what are the use-cases for supporting the async behavior?